okhttp/src/main/java/okhttp3/RealCall.java

```diff
@@ -33,7 +33,10 @@
 final class RealCall implements Call {
   final OkHttpClient client;
   final RetryAndFollowUpInterceptor retryAndFollowUpInterceptor;
-  final EventListener eventListener;
+
+  // There is a cycle between the Call and EventListener that makes this awkward. This will be set
+  // after we create the call instance then create the event listener instance.
+  volatile EventListener eventListener;
```

Comment:
This code is single-threaded, so volatile and safe-publication aren’t relevant?

The `call.execute()` line is going to not have an event listener anyway because that isn’t known yet. So the `call.execute()` will throw when it gets to the point that it’s trying to use its event listener.

We should document that it is an error for an EventListener to perform any mutating operation on the Call – heck, we could potentially even enforce that. In either case, it’s definitely not sane for an event listener to execute a call; that’s not its responsibility.